main.py

```
@@ -0,0 +1,165 @@
+class SearchRequest(BaseModel):
+    query: str = Field(..., min_length=1, description="The search query.")
+
+class SearchResult(BaseModel):
+    title: str
+    link: str
+    snippet: str
+
+@app.post("/tools/search", response_model=list[SearchResult], tags=["Tools"])
+def execute_search(
+    search_request: SearchRequest,
+    settings: Settings = Depends(get_settings)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)